### PR TITLE
sanctuary-type-classes@5.2.0

### PR DESCRIPTION
**Pull request by davidchambers**
<https://github.com/sanctuary-js/sanctuary-type-classes/releases/tag/v5.2.0>

<del>I'd like to wait until sanctuary-js/sanctuary-type-classes#56 is merged before releasing `sanctuary-type-classes@5.2.0`, at which point I will update this pull request.</del>

### CODE REVIEW

**Comment by gabejohnson** [comment]
`S.I` is equivalent to `Z.id(Function)`. What are some reasons _not_ to define `S.I` in terms of `Z.id`?

**Comment by davidchambers** [comment]
> What are some reasons *not* to define `S.I` in terms of `Z.id`?

The one that occurs to me is function hoisting. Because Sanctuary's internal functions are defined via function declarations we are free to use one internal function in the body of another without worrying about ordering. Replacing `function I(x) { return x; }` with `var I = Z.id(Function);` would make ordering significant.

**Comment by gabejohnson** [comment]
It doesn't appear that `I` is used internally anywhere. Thus `S.I` could be defined
```js
S.I = def('I', {}, [a, a], Z.id(Function));
```
But I suppose there's currently no benefit in doing so.

File: index.js [hunk 1]
```diff
@@ -1492,23 +1503,27 @@
...
   //. ```
-  function compose(f, g, x) {
-    return f(g(x));
-  }
-  S.compose = def('compose', {}, [Fn(b, c), Fn(a, b), a, c], compose);
+  S.compose =
+  def('compose',
+      {s: [Z.Semigroupoid]},
+      [s(b, c), s(a, b), s(a, c)],
+      Z.compose);
```
Comment (gabejohnson) [inline index.js]: This is a breaking change. I didn't realize the function is currently ternary. It was surprising.

**Comment by davidchambers** [comment]
> It doesn't appear that `I` is used internally anywhere.

I just used `I` in #397. ;)

File: index.js (hunk 1 above)
Comment (davidchambers) [inline index.js]: My policy is to accept as many arguments as possible. :)

**Pull request merged**